src/http.c

```diff
@@ -77,6 +77,24 @@ int http_connect( http_t *conn, int proto, char *proxy, char *host, int port, ch
 		conn->local_if, conn->headers ) == -1 )
 		return( 0 );
 
+	//For SSl through proxy
+	if( conn->proxy && PROTO_IS_SECURE(conn->proto))
+	  {
+	    char proxybuffer[2000];
```

Comment:
Please use constants instead of numbers. Perhaps 
#define USER_AGENT_LEN 2000.
Also, it would be nice to have the rationale behind the size (2000) selection documented.